pandas/core/reshape/melt.py

```diff
@@ -32,7 +33,12 @@ def melt(frame, id_vars=None, value_vars=None, var_name=None,
             raise ValueError('id_vars must be a list of tuples when columns'
                              ' are a MultiIndex')
         else:
+            # Check that `id_vars` are in frame
             id_vars = list(id_vars)
+            missing = Index(id_vars).difference(frame.columns)
+            if not missing.empty:
+                raise ValueError('Columns {missing} are not in'
```

Comment:
Should this be a KeyError instead?

Reply:
Put single-quotes around the `{missing}`. What's the exact error message? I think you should convert the Index to a list so that `Index([...])` isn't in the error message, just the extra column names.

Reply:
Same two comments apply below as well.

Reply:
Can you also note that this is for the `id_vars` parameter? Something like

```
Some items in 'id_vars' not present in the DataFrame: '{missing}'
```